### ISSUE TITLE
Cloudflare DNS: Unable to verify - Challenge is not correct

### ISSUE THREAD

**Issue by ChaosZero112** [webprofusion/certify#352]
Hello,

I've tried searching through the closed issues for a solution to my problem, but so far the solutions I've seen have had no effect.

I'm unable to verify any domains using Cloudflare API. The logs always show the correct record found, but it thinks it's the wrong one.

It doesn't matter if it's 1 domain, SAN or wildcard. Same result.

Client version is 4.0.5.

Logs attached,

[log_0b4e8446-f7f9-4bfc-aa4a-cacca29b2932_1.txt](https://github.com/webprofusion/certify/files/2262900/log_0b4e8446-f7f9-4bfc-aa4a-cacca29b2932_1.txt)

```
2018-08-06 10:12:55.567 -04:00 [INF] Attempting Domain Validation: [domain].com
2018-08-06 10:12:55.579 -04:00 [INF] Registering and Validating [domain].com 
2018-08-06 10:12:55.579 -04:00 [INF] Performing automated challenge responses ([domain].com)
2018-08-06 10:12:55.626 -04:00 [INF] DNS: Creating TXT Record '_acme-challenge.[domain].com' with value '**bySjbh6FZxQXy4pBvDx4CC2tP6Nbc2jF39ynYko7X1Q**', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-06 10:12:56.388 -04:00 [INF] DNS: Cloudflare DNS API :: DNS record added: _acme-challenge.[domain].com
2018-08-06 10:14:03.443 -04:00 [INF] Attempting Challenge Response Validation for Domain: [domain].com
2018-08-06 10:14:03.459 -04:00 [INF] Registering and Validating [domain].com 
2018-08-06 10:14:03.459 -04:00 [INF] Checking automated challenge response for Domain: [domain].com
2018-08-06 10:14:04.687 -04:00 [INF] Incorrect TXT record "**bySjbh6FZxQXy4pBvDx4CC2tP6Nbc2jF39ynYko7X1Q**" found at _acme-challenge.[domain].com
```

**Comment by webprofusion-chrisc** [comment]
Hi Cody, thanks for raising this. I'm not aware of any problems with the cloudflare DNS provider currently. Can you try deleting the existing TXT record from your DNS zone manually, then wait a few minutes and try your request again. I'm trying to determine if the propagation time between your nameservers is slow. Can you also try out https://letsdebug.net/ to see what it says.

**Comment by ChaosZero112** [comment]
Hi @webprofusion-chrisc , thank you for the reply.
I cleaned all TXT records on the domain and got the same error. What I did notice is the program placed 2 validation records in the same place and tried to verify them at the same time.

![Cloudflare](http://tinyimg.io/i/pwKIqNg.png)

`2018-08-09 10:09:05.902 -04:00 [INF] Attempting Domain Validation: *.[domain].com
2018-08-09 10:09:05.902 -04:00 [INF] Registering and Validating *.[domain].com 
2018-08-09 10:09:05.902 -04:00 [INF] Performing automated challenge responses (*.[domain].com)
2018-08-09 10:09:05.907 -04:00 [INF] DNS: Creating TXT Record '_acme-challenge.[domain].com' with value 'X_oYdGgmLt_VZah2xasPkACLKwN3p2p_S4X6VCHEAzQ', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-09 10:09:06.935 -04:00 [INF] DNS: Cloudflare DNS API :: DNS record added: _acme-challenge.[domain].com
2018-08-09 10:09:06.936 -04:00 [INF] Requesting Validation from Let's Encrypt: *.[domain].com
2018-08-09 10:09:06.936 -04:00 [INF] Attempting Domain Validation: [domain].com
2018-08-09 10:09:06.936 -04:00 [INF] Registering and Validating [domain].com 
2018-08-09 10:09:06.938 -04:00 [INF] Performing automated challenge responses ([domain].com)
2018-08-09 10:09:06.942 -04:00 [INF] DNS: Creating TXT Record '_acme-challenge.[domain].com' with value 'HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-09 10:09:08.060 -04:00 [INF] DNS: Cloudflare DNS API :: DNS record added: _acme-challenge.[domain].com
2018-08-09 10:09:08.060 -04:00 [INF] Requesting Validation from Let's Encrypt: [domain].com
2018-08-09 10:10:08.554 -04:00 [INF] Attempting Challenge Response Validation for Domain: *.[domain].com
2018-08-09 10:10:08.554 -04:00 [INF] Registering and Validating *.[domain].com 
2018-08-09 10:10:08.554 -04:00 [INF] Checking automated challenge response for Domain: *.[domain].com
2018-08-09 10:10:09.362 -04:00 [INF] Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" (and 1 more) found at _acme-challenge.[domain].com
2018-08-09 10:10:11.268 -04:00 [INF] DNS: Deleting TXT Record '_acme-challenge.[domain].com', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-09 10:10:11.925 -04:00 [INF] Validation of the required challenges did not complete successfully. Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" (and 1 more) found at _acme-challenge.[domain].com
2018-08-09 10:10:11.925 -04:00 [INF] Validation of the required challenges did not complete successfully. Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" (and 1 more) found at _acme-challenge.[domain].com`

Just to be sure, I also tried to validate just a single domain without the wildcard and got the same result.

`2018-08-09 10:22:55.313 -04:00 [INF] All Tests Completed OK
2018-08-09 10:23:03.510 -04:00 [INF] Beginning Certificate Request Process: [domain].com using ACME Provider:Certes
2018-08-09 10:23:03.510 -04:00 [INF] Registering Domain Identifiers
2018-08-09 10:23:04.966 -04:00 [INF] Created ACME Order: https://acme-v02.api.letsencrypt.org/acme/order/38349660/33838673
2018-08-09 10:23:05.163 -04:00 [VRB] Fetching Authorizations.
2018-08-09 10:23:05.312 -04:00 [VRB] Fetching Authz Challenges.
2018-08-09 10:23:05.980 -04:00 [INF] Got http-01 challenge https://acme-v02.api.letsencrypt.org/acme/challenge/L79-51i5sdDQCivKkRhwoib9_d0KnnBp-bCYY6tuo5c/6313167084
2018-08-09 10:23:06.410 -04:00 [INF] Got dns-01 challenge https://acme-v02.api.letsencrypt.org/acme/challenge/L79-51i5sdDQCivKkRhwoib9_d0KnnBp-bCYY6tuo5c/6313167085
2018-08-09 10:23:06.410 -04:00 [INF] Attempting Domain Validation: [domain].com
2018-08-09 10:23:06.411 -04:00 [INF] Registering and Validating [domain].com 
2018-08-09 10:23:06.411 -04:00 [INF] Performing automated challenge responses ([domain].com)
2018-08-09 10:23:06.416 -04:00 [INF] DNS: Creating TXT Record '_acme-challenge.[domain].com' with value 'HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-09 10:23:07.089 -04:00 [INF] DNS: Cloudflare DNS API :: DNS record added: _acme-challenge.[domain].com
2018-08-09 10:23:07.089 -04:00 [INF] Requesting Validation from Let's Encrypt: [domain].com
2018-08-09 10:24:07.607 -04:00 [INF] Attempting Challenge Response Validation for Domain: [domain].com
2018-08-09 10:24:07.607 -04:00 [INF] Registering and Validating [domain].com 
2018-08-09 10:24:07.608 -04:00 [INF] Checking automated challenge response for Domain: [domain].com
2018-08-09 10:24:08.414 -04:00 [INF] Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" found at _acme-challenge.[domain].com
2018-08-09 10:24:09.562 -04:00 [INF] DNS: Deleting TXT Record '_acme-challenge.[domain].com', in Zone Id '275a08d57f81f05bd0b62108963f4fbe' using API provider 'Cloudflare DNS API'
2018-08-09 10:24:10.256 -04:00 [INF] Validation of the required challenges did not complete successfully. Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" found at _acme-challenge.[domain].com
2018-08-09 10:24:10.256 -04:00 [INF] Validation of the required challenges did not complete successfully. Incorrect TXT record "HwISKHnZYDmp4gIhY115gLPSluYPhlpof17JM78NdcE" found at _acme-challenge.[domain].com`


It should also be noted I had no issue using Posh-ACME to obtain the wildcard.
